lib/vanagon/component/dsl.rb

```diff
@@ -25,6 +25,30 @@ def component(name, &block)
         yield(self, @component.settings, @component.platform)
       end
 
+      # Define a gem component with default values. Automatically sets `version`, `url`, `build_requires`,
+      # and `install`. Fixes up the PATH for Windows. Requires `ruby_version`, `gem_path_env`, and
+      # `gem_install` settings.
+      #
+      # @param name [String] name of the gem
+      # @param version [String] name of the componennt
+      # @param block [Proc] DSL definition of the component to call
+      def gem(name, version, &block)
+        self.version version
+        url "https://rubygems.org/gems/#{name}-#{version}.gem"
```

Comment:
I think that this will set the component source type's URL as an [HTTP component](https://github.com/puppetlabs/vanagon/blob/master/lib/vanagon/component/source.rb#L27-L34), which should also then require a checksum. Does that not happen?